lib/mayaUsd/utils/utilFileSystem.cpp

```diff
@@ -117,6 +117,14 @@ std::string UsdMayaUtilFileSystem::getMayaSceneFileDir()
     return std::string();
 }
 
+std::string UsdMayaUtilFileSystem::getPathRelativeToMayaSceneFile(const std::string& fileName)
+{
+    ghc::filesystem::path absolutePath(fileName);
+    ghc::filesystem::path basePath(getMayaSceneFileDir());
+    ghc::filesystem::path relativePath = absolutePath.lexically_relative(basePath);
+    return relativePath.generic_string();
```

Comment:
Note that `lexically_relative` can return an empty path is multiple corner cases, for example if both path are entirely separate, for example in Windows if each path is on a different drive (example: C:, D:).

IDK how we want to handle the case where the user requested relative paths and they cannot be made relative: do we show a warning, an error?

At the very least, we probably want to return the original fileName in that case.

Reply:
See: https://en.cppreference.com/w/cpp/filesystem/path/lexically_normal